### PR TITLE
Fixes the Write-Pode(*)Response functions so the Value parameter appropriately handles when an array is passed using piping

### PR DESCRIPTION
**Pull request by mdaneri**
Address the #1272 bug (Write-PodeHtmlResponse returns the wrong result when an array of strings is piped.)
The same issue is present with the following functions:

- Write-PodeCsvResponse
- Write-PodeHtmlResponse
- Write-PodeJsonResponse
- Write-PodeXmlResponse
- Write-PodeYamlResponse
 

 

### Description of the Change
Added support for array values passed using piping

### Related Issue
Issue @1272

### Examples
```
Add-PodeRoute -Path '/processesPipedNotWorking'  -Method Get -ScriptBlock {
        Get-Process | .{ process { if ($_.WS -gt 100mb) { $_ } } } |
            Select-Object Name, @{e = { [int]($_.WS / 1mb) }; n = 'WS' } |
            Sort-Object WS -Descending | Write-PodeHtmlResponse  -StatusCode 200
    }
```

### CODE REVIEW

File: src/Public/Responses.ps1 [hunk 1]
```diff
@@ -830,30 +908,48 @@ function Write-PodeYamlResponse {
...
+            'value' {
+                if ($pipelineValue) {
+                    $Value = $pipelineValue
+                }
+
+                if ($Value -is [hashtable]) {
+                    $Value = @(foreach ($v in $Value) {
+                            New-Object psobject -Property $v
+                        })
+                }
```
Comment (Badgerati) [inline src/Public/Responses.ps1:939]: Is this needed for YAML? Because `ConvertTo-PodeYamlInternal` handles hashtables 🤔

**Review by Badgerati — CHANGES_REQUESTED** [review]

File: src/Public/Responses.ps1 (hunk 1 above)
Comment (mdaneri) [reply src/Public/Responses.ps1:939]: no yaml doesn't need it like json.
only CSV and XML need it

Comment (mdaneri) [reply src/Public/Responses.ps1:939]: this was a refuse from my tests.
it's useless and it's not required

**Review by Badgerati — APPROVED** [review]
LGTM

**Pull request merged**